### ISSUE TITLE
different behavior of rel_entr

### ISSUE THREAD

**Issue by sh0416** [google/jax#21192]
### Description

```
import jax
import jax.numpy as jnp
from scipy.special import rel_entr

p = jnp.array([0.7, 0.3, 0.0, 0.0])
q = jnp.array([0.0, 0.3, 0.7, 0.0])
print(jax.scipy.special.rel_entr(p, q))
p = np.asarray(p)
q = np.asarray(q)
print(rel_entr(p, q))
```

shows

```
[       inf 0.00000000 0.69999999 0.00000000]  # jax result
[       inf 0.00000000 0.00000000 0.00000000]  # scipy result
```

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.4.25
jaxlib: 0.4.25
numpy:  1.26.4
python: 3.10.13 (main, Sep 11 2023, 13:44:35) [GCC 11.2.0]
jax.devices (1 total, 1 local): [cuda(id=0)]
process_count: 1
platform: uname_result(system='Linux', node='gpusystem', release='5.15.0-100-generic', version='#110-Ubuntu SMP Wed Feb 7 13:27:48 UTC 2024', machine='x86_64')
```

**Comment by rajasekharporeddy** [comment]
Hi @sh0416 

I think this issue can now be closed since the PR #21193 was already merged. Also, the fix is now available in JAX public release version 0.4.29.

Thank you.

**Comment by sh0416** [comment]
Thanks for the notice. Close issue.
